### PR TITLE
Meta with services

### PR DESCRIPTION
**Pull request by WouterSioen**
This makes sure we can make a "repository" class which isn't static. If
the meta class finds this service in the container, it will use it
instead of static method calls.

### CODE REVIEW

**Comment by WouterSioen** [comment]
Example usage:

        $this->meta->setURLCallback('job_repository', 'getUrl');

Where `job_repository` is a service which contains a `getUrl` method.

**Comment by WouterSioen** [comment]
@jessedobbelaere this could probably be useful for you, since you use repository services (and probably already encountered this issue).

Can you check this if you have time?

**Comment by WouterSioen** [comment]
Note that I'm already using it in a project, so it probably works well ;)

File: src/Backend/Core/Engine/Meta.php [hunk 1]
```diff
@@ -471,39 +478,16 @@ public function save($update = false)
...
-        if ($this->frm->getField('seo_index')->getValue() != 'none') {
-            $meta['data']['seo_index'] = $this->frm->getField('seo_index')->getValue();
-        }
-        if ($this->frm->getField('seo_follow')->getValue() != 'none') {
-            $meta['data']['seo_follow'] = $this->frm->getField('seo_follow')->getValue();
-        }
-        if (isset($meta['data'])) {
-            $meta['data'] = serialize($meta['data']);
-        }
-
```
Comment (jeroendesloovere) [inline src/Backend/Core/Engine/Meta.php]: I see that you aren't setting the $meta parameter anymore.
Where has this gone to?

Comment (WouterSioen) [inline src/Backend/Core/Engine/Meta.php]: this was done twice

**Pull request merged**